### DISCUSSION TITLE
Arch freezing / reboots on Surface Pro 8 solved by setting `split_lock_detect=off`

### DISCUSSION THREAD

**Discussion by snide** [linux-surface/linux-surface#1132] [General]
Ran into an issue with my dual boot Surface Pro 8 that is running Arch + Gnome where everything was working, but it would constantly freeze, and then eventually crash. Looking at the crash logs I kept seeing that it was something to do with a linux parameter called `split_lock_detect`. Turning that to `off` in your `grub` startup seemed to completely fix the issue for me. It seems this is common when folks are running Windows in a VM, so I'm guessing something with the dual boot may be at play, but figured I'd post here so other folks have something to google. I should note that I upgraded my hard drive to 1TB, and had similar freezing problems in Windows until I started adjusting the power parameters there.

**Comment by qzed** [comment]
The warning means that something (as in either Windows itself, which I doubt, or an application running on either Linux or in the Windows VM) is trying to execute an atomic operation (like atomic-add/-sub/...) that spans across two cache-lines. The problem with that is that as soon as an atomic operation spans more than one cache line, the whole bus needs to be locked, which has performance implications on not just the problematic application. Due to this, the kernel devs decided at some point to "punish" the offending application by slowing it down (I believe this is off by default) and warning about it. You can find more details about that [here](https://docs.kernel.org/x86/buslock.html).

So ideally, whatever application causes this needs to be fixed. Given that it's very likely a Windows application, however, I don't see that happening. So I would say that on normal end-user machines turning that off is a valid option as well (as in: there should be no noticeable downsides for you).
